datafusion/sql/src/expr/value.rs

```diff
@@ -151,16 +151,51 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let data_types: HashSet<DataType> =
             values.iter().map(|e| e.get_datatype()).collect();
 
-        if data_types.is_empty() {
-            Ok(lit(ScalarValue::new_list(None, DataType::Utf8)))
-        } else if data_types.len() > 1 {
-            Err(DataFusionError::NotImplemented(format!(
-                "Arrays with different types are not supported: {data_types:?}",
-            )))
-        } else {
-            let data_type = values[0].get_datatype();
+        match data_types.len() {
+            0 => Ok(lit(ScalarValue::new_list(None, DataType::Utf8))),
+            1 => {
+                let data_type = values[0].get_datatype();
+
+                Ok(lit(ScalarValue::new_list(Some(values), data_type)))
+            }
+            // Check whether one of them is null
+            2 => {
+                let has_null_type = data_types.contains(&DataType::Null);
+                if has_null_type {
+                    // convert to Vec for indexing
+                    let data_types = data_types.into_iter().collect::<Vec<_>>();
+                    let non_null_type = if data_types[0] == DataType::Null {
```

Comment:
How do we know the Null is the first or second type? Maybe you could use something like 

```
let non_null_type = data_types.find(|d| d == DataType::Null)
```

or something

Reply:
I assume there is only null and `one` non-null, if not, this fails